libs/counterfactuals/src/util/getFetchPredictionPromise.ts

```diff
@@ -6,14 +6,23 @@ import { JointDataset } from "@responsible-ai/core-ui";
 export function getFetchPredictionPromise(
   fetchingReference: { [key: string]: any },
   jointDataset: JointDataset,
-  invokeModel: (data: any[], abortSignal: AbortSignal) => Promise<any[]>
+  invokeModel: (data: any[], abortSignal: AbortSignal) => Promise<any[]>,
+  testData?: any[],
+  ifEnableLargeData?: boolean
 ): Promise<any[]> {
   const abortController = new AbortController();
-  const rawData = JointDataset.datasetSlice(
-    fetchingReference,
-    jointDataset.metaDict,
-    jointDataset.datasetFeatureCount
-  );
-  const promise = invokeModel([rawData], abortController.signal);
+  let rawData = [
+    JointDataset.datasetSlice(
+      fetchingReference,
+      jointDataset.metaDict,
+      jointDataset.datasetFeatureCount
+    )
+  ];
+  if (ifEnableLargeData && testData) {
+    testData[0][0].pop(); // drop the prediction value before predict call
```

Comment:
Since we are mutating this value might this affect any visualizations that might render during this call - or do we do a deep copy of testData before this somewhere/not use it directly in visualizations?

Reply:
Should this be the last value in the list? Could you plese confirm?

Reply:
Thanks @imatiach-msft !! we are using using test data to compare and display data in panel. I updated to use cloned data to be used in predict call.
@gaugup yes, I am just removing last value from the list.